### PR TITLE
docs: add thresholds example

### PR DESCRIPTION
**Pull request by hugorut**
* adds simple thresholds example to `examples` folder

---

Tested and working on local atlantis runner

### CODE REVIEW

File: examples/thresholds/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+               absolute_percent_change=$(echo "${percent_change#-}")
+               
+               echo "past: ${past}"
+               echo "current: ${current}"
+               echo "cost_change: ${cost_change}"
+               echo "absolute_cost_change: ${cost_change#-}"
+               echo "percent_change: ${percent_change}"
+               echo "absolute_percent_change: ${absolute_percent_change}"
+   
+               if [ $(echo "$percent_change < 1" | bc -l) == "1" ]; then           # Only comment if cost changed by more than plus or minus 1%
```
Comment (hugorut) [inline examples/thresholds/README.md:100]: this is funky but is the only way I could get it to evaluate correctly as these numbers can be floats.

File: examples/thresholds/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+This example shows you how to run Infracost with Atlantis to post a comment on your PR when cloud costs exceed set thresholds.
+
+## Running with GitHub
+
+1. Update your setup to use the [infracost-atlantis](https://hub.docker.com/r/infracost/infracost-atlantis) dockerhub image
```
Comment (aliscott) [inline examples/thresholds/README.md:5]:
```suggestion
1. Update your setup to use the [infracost-atlantis](https://hub.docker.com/r/infracost/infracost-atlantis) Docker image
```

File: examples/thresholds/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+     terraform-infracost:
+       plan:
+         steps:
+           - init
+           - plan
+           # Run Infracost breakdown and save to a tempfile, namespaced by this project, PR, workspace and dir
+           - run: infracost breakdown --path=$PLANFILE --format=json --log-level=info --out-file=/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json
+           - run: |
+               # Read the breakdown JSON and get costs
+               past=$(cat /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json | jq -r "(.pastTotalMonthlyCost // 0) | tonumber")
```
Comment (aliscott) [inline examples/thresholds/README.md:26]: This examples makes me wonder if there's any way we can make this file path shorter? Unfortunately we can't use `$PROJECT_NAME` because that's not always set. Is it possible to set an env var to that value and use that?

I think this is outside of the scope of this PR though so happy to look at it separately.

Comment (aliscott) [reply examples/thresholds/README.md:26]: This will work with the infracost-atlantis docker image, but not with the "Install in Docker" option (https://github.com/infracost/infracost-atlantis/tree/hail_atlantis#option-2-install-in-docker) since `jq` isn't included in the Atlantis docker image. Maybe all we need for this is a note at the top to show how to install `jq` if it's not already installed.

File: examples/thresholds/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+               # if [ $(echo "$absolute_cost_change < 100" | bc -l) == "1" ]; then # Only comment if cost changed by more than plus or minus $100
+               # if [ $(echo "$cost_change < 100" | bc -l) == "1" ]; then          # Only comment if cost increased by more than $100
+                 echo "Skipping comment, absolute percentage change is less than 1"
+                 exit 0
+               fi
+   
+               infracost comment github --repo $BASE_REPO_OWNER/$BASE_REPO_NAME --pull-request $PULL_NUM --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json --github-token $GITHUB_TOKEN
+   ```
+4. Restart the atlantis application with the new env vars and config
+5. Send a pull request in GitHub to change something in TF, the Infracost pull request comment will be added when you go above your set threshold.
```
Comment (aliscott) [inline examples/thresholds/README.md:56]:
```suggestion
4. Restart the Atlantis application with the new env vars and config
5. Send a pull request in GitHub to change something in Terraform code, the Infracost pull request comment will be added when you go above your set threshold.
```

File: examples/thresholds/README.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+## Running with Gitlab
+
+1. Update your setup to use the [infracost-atlantis](https://hub.docker.com/r/infracost/infracost-atlantis) dockerhub image
```
Comment (aliscott) [inline examples/thresholds/README.md:60]:
```suggestion
1. Update your setup to use the [infracost-atlantis](https://hub.docker.com/r/infracost/infracost-atlantis) Docker image
```

File: examples/thresholds/README.md (hunk 1 above)
Comment (aliscott) [reply examples/thresholds/README.md:100]: Should we use a different number other than `1` for the percentage change to make it clearer which is which? Or move that into an variable called `threshold` or something?

File: examples/thresholds/README.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+               # if [ $(echo "$absolute_cost_change < 100" | bc -l) == "1" ]; then # Only comment if cost changed by more than plus or minus $100
+               # if [ $(echo "$cost_change < 100" | bc -l) == "1" ]; then          # Only comment if cost increased by more than $100
+                 echo "Skipping comment, absolute percentage change is less than 1"
+                 exit 0
+               fi
+   
+               infracost comment gitlab --repo $BASE_REPO_OWNER/$BASE_REPO_NAME --merge-request $PULL_NUM --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM-$WORKSPACE-$REPO_REL_DIR-infracost.json --gitlab-token $GITLAB_TOKEN
+   ```
+4. Restart the atlantis application with the new env vars and config
+5. Send a merge request in GitLab to change something in TF, the Infracost merge request comment will be added when you go above your set threshold.
```
Comment (aliscott) [inline examples/thresholds/README.md:111]:
```suggestion
4. Restart the Atlantis application with the new env vars and config
5. Send a merge request in GitLab to change something in the Terraform code, the Infracost merge request comment will be added when you go above your set threshold.
```

**Review by aliscott — APPROVED** [review]
🚀 🚀  bash guru

File: examples/thresholds/README.md (hunk 3 above)
Comment (hugorut) [reply examples/thresholds/README.md:26]: > This examples makes me wonder if there's any way we can make this file path shorter? Unfortunately we can't use `$PROJECT_NAME` because that's not always set. Is it possible to set an env var to that value and use that?

yeah good idea at setting an env var - since this affects most other examples, let merge this and resolve all others

Comment (hugorut) [reply examples/thresholds/README.md:26]: > This will work with the infracost-atlantis docker image, but not with the "Install in Docker" option (https://github.com/infracost/infracost-atlantis/tree/hail_atlantis#option-2-install-in-docker) since `jq` isn't included in the Atlantis docker image. Maybe all we need for this is a note at the top to show how to install `jq` if it's not already installed.

I'll add a note

File: examples/thresholds/README.md (hunk 1 above)
Comment (hugorut) [reply examples/thresholds/README.md:100]: 👍
